samples/js/photos/src/photo-grid-item.js

```diff
@@ -43,53 +51,95 @@ export default class PhotoGridItem extends React.Component<void, Props, State> {
   constructor(props: Props) {
     super(props);
     this.state = {
-      size: props.gridSize,
-      sizeIsBest: false,
-      url: props.url,
+      blob: this._getBlob(props),
+      url: null,
     };
     this._parentTop = 0;
     this._parentLeft = 0;
-    this._shouldTransition = true;
+    this._load(this.state.blob);
   }
 
-  componentWillUpdate() {
+  componentWillReceiveProps(nextProps: Props) {
+    const nextBlob = this._getBlob(nextProps);
+    if (nextBlob === this.state.blob) {
+      return;
+    }
+    this.setState({
+      blob: nextBlob,
+    });
+    this._load(nextBlob);
+  }
+
+  componentWillUpdate(nextProps: Props, nextState: State) {
+    if (this.state.url !== nextState.url) {
```

Comment:
It seems like it would be better to do this at the point the URL state actually updates, i.e. from _load.